docs/contributing/hacking.rst

```diff
@@ -36,41 +69,108 @@ docker container. To do that, run
     $ ./make.bash --no-clean
     $ GOOS=linux GOARCH=amd64 ./make.bash --no-clean
 
-After that, you should be able to compile Deis' components as normal.
+Once you can compile to ``linux/amd64``, you should be able to compile Deis'
+components as normal.
+
+Configuring Boot2Docker (Mac)
+`````````````````````````````
+
+Deis needs a Docker registry running independently of the Deis cluster. On
+OS X, you will need to run this docker registry inside of Boot2Docker (http://boot2docker.io).
+
+Install Boot2Docker according to the normal installation instructions. When you
+run ``init``, we highly recommend allocating a large disk, since the Docker
+registry that will live there is fairly large.
+
+.. code-block:: console
+
+    $ boot2docker init --disksize=100000
+
+That will create virtual disk that can eventually take up a full 100,000MB of
+disk space. Then start up Boot2Docker.
+
+Once you have run ``boot2docker up``, you should be able to connect to it. You
+need to make a minor editor to the boot2docker config:
+
+.. code-block:: console
+
+    $ boot2docker ssh sudo vi /var/lib/boot2docker/profile
+
+Inside of the profile, you need to add one line:
+
+.. code-block:: console
+
+    EXTRA_ARGS="--insecure-registry 192.168.59.103:5000"
+
+Once that line has been added, you can either restart boot2docker's docker
+server, or you can restart boot2docker. We recommend the later.
+
+.. code-block:: console
 
-The development workflow requires a Docker Registry that is accessible to you
-(the developer) and to all of the hosts in your cluster.
+    $ boot2docker halt
+    $ boot2docker up
 
-You will also need a `deisctl`_ client to update images and restart components.
+At this point, Boot2Docker can now serve as a registry for Deis' Docker images.
+Later on we will return to this.
 
-Fork the Repository
--------------------
+Fork the Deis Repository
+------------------------
+Once the prerequisites have been met, we can begin to work with Deis.
 
 To get Deis running for development, first `fork the Deis repository`_,
-then clone your fork of the repository:
+then clone your fork of the repository. Since Deis is predominantly written
+in Go, the best place to put it is in ``$GOPATH/src/github.com/deis/``
 
 .. code-block:: console
 
-	$ git clone git@github.com:<username>/deis.git
-	$ cd deis
+    $ mkdir -p  $GOPATH/src/github.com/deis
+    $ cd $GOPATH/src/github.com/deis
+    $ git clone git@github.com:<username>/deis.git
```

Comment:
Indeed. The pushurl workflow doesn't work for me because I work across three different machines at the same time, so I need to pull from both local and upstream repositories as well as pushing to more than one remote upstream (I manage both my personal org as well as fishwork's org). I'm happy with this approach, but others are welcome to use the other approach if they feel the need. Its a personal preference thing.